src/main/java/org/mutabilitydetector/checkers/AsmMutabilityChecker.java

```diff
@@ -42,11 +42,20 @@
     private CheckerResult checkerResult = IMMUTABLE_CHECKER_RESULT;
 
     protected Collection<MutableReasonDetail> reasons = newArrayList();
-
+    
+    private boolean isClassSelfReferenced=false;
+    
     public String ownerClass() {
         return ownerClass;
     }
 
+    protected void setIsClassSelfReferenced(boolean isClassSeflReferenced) {
+        this.isClassSelfReferenced = isClassSeflReferenced;
+    }
+    
+    protected Boolean isClassSelfReferenced() {
+        return new Boolean(isClassSelfReferenced);
+    }
```

Comment:
The boxing doesn't seem necessary, thus:
```java
protected boolean isClassSelfReferenced() {
    return isClassSelfReferenced;
}
```
Idem further down.

Reply:
done, yeah not sure what was i thinking when i was boxing it to Boolean object